### PR TITLE
added prettier to the editor

### PR DESCRIPTION
**Pull request by ggrandi**
so far I have a added a sensible prettier config and bumped the prettier version

next steps:
- Add the ability to change the prettier config (gfi)

### CODE REVIEW

File: packages/monaco/src/prettier.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import estree from 'prettier/plugins/estree';
+import typescript from 'prettier/plugins/typescript';
+import type { Options } from 'prettier';
+import type * as monaco from 'monaco-editor';
+
+// TODO: use the settings store for the config
+const prettierConfig = {
+  plugins: [estree, typescript],
+  parser: 'typescript',
+  useTabs: true,
```
Comment (Hacksore) [inline packages/monaco/src/prettier.ts:11]: controversial setting 😬 

Maybe you could make this use the current editor setting?

**Review by Hacksore — APPROVED** [review]

File: packages/monaco/src/prettier.ts (hunk 1 above)
Comment (ggrandi) [reply packages/monaco/src/prettier.ts:11]: I think we already control the tab size with the `settings-store` so it shouldn't be an issue

Plus this saves us many bytes of space in the db

Comment (Hacksore) [reply packages/monaco/src/prettier.ts:11]: oh we don't even support spaces lmao
image

Comment (ggrandi) [reply packages/monaco/src/prettier.ts:11]: I think we support whatever the editor uses which might default to spaces

**Pull request merged**